acceptance/detector_test.go

```diff
@@ -0,0 +1,197 @@
+	defer h.DockerImageRemove(t, detectImage)
+
+	spec.Run(t, "acceptance-detector", testDetector, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testDetector(t *testing.T, when spec.G, it spec.S) {
+	when("called with arguments", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", detectImage, "some-arg")
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to parse arguments: received unexpected arguments"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("running as a root", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", "--user", "root", detectImage)
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to build: refusing to run as root"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("read buildpack order file failed", func() {
+		it("errors", func() {
+			// no order.toml file in the default directory
+			cmd := exec.Command("docker", "run", "--rm", "--user", userID, detectImage)
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to read buildpack order file"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("no buildpack group passed detection", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", "--user", userID, "--env", "CNB_ORDER_PATH=/cnb/orders/empty_order.toml", detectImage)
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "No buildpack groups passed detection."
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("there is a buildpack group that pass detection", func() {
+		var copyDir, containerName string
+
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
+		})
+
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			os.RemoveAll(copyDir)
+		})
+
+		it("writes group.toml and plan.toml", func() {
+			h.DockerRunAndCopy(t,
+				containerName,
+				copyDir,
+				detectImage,
+				"/layers",
+				h.WithFlags("--user", userID,
+					"--env", "CNB_ORDER_PATH=/cnb/orders/simple_order.toml",
+				),
+				h.WithArgs(),
+			)
+
+			// check group.toml
+			tempGroupToml := filepath.Join(copyDir, "layers", "group.toml")
+			groupContents, err := ioutil.ReadFile(tempGroupToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(groupContents) > 0, true)
+			var buildpackGroup lifecycle.BuildpackGroup
+			_, err = toml.Decode(string(groupContents), &buildpackGroup)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildpackGroup.Group[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildpackGroup.Group[0].Version, "simple_buildpack_version")
+
+			// check plan.toml - should be empty
```

Comment:
This comment doesn't seem to apply anymore?

Reply:
oops :) I copied it from the next test. Thanks!